deepeval/metrics/hallucination_evaluation.py

```diff
@@ -0,0 +1,64 @@
+import os
+from deepeval.singleton import Singleton
+from deepeval.test_case import LLMTestCase
+from deepeval.utils import chunk_text, softmax
+from deepeval.metrics.base_metric import BaseMetric
+from deepeval.evaluator import assert_test
+from deepeval.progress_context import progress_context
+from sentence_transformers import CrossEncoder
+
+
+class HallucinationEvaluationModel(metaclass=Singleton):
+    def __init__(
+        self, model_name: str = "vectara/hallucination_evaluation_model"
+    ):
+        # We use a smple cross encoder model
+        os.environ["TOKENIZERS_PARALLELISM"] = "false"
+        self.model = CrossEncoder(model_name)
+
+    def predict(self, text_a: str, text_b: str):
+        scores = self.model.predict([[text_a, text_b]])
+        # https://huggingface.co/vectara/hallucination_evaluation_model
+
+        return scores[0]
+
+
+class HallucinationMetric(BaseMetric, metaclass=Singleton):
+    def __init__(
+        self,
+        minimum_score: float = 0.5,
+        model_name: str = "vectara/hallucination_evaluation_model",
+    ):
+        # For Crossencoder model, move to singleton to avoid re-instantiating
+
+        with progress_context(
+            "Downloading HallucinationEvaluationModel (may take up to 2 minutes if running for the first time)..."
+        ):
+            self.model = VectaraHallucinationEvaluationModel(model_name)
+        self.minimum_score = minimum_score
```

Comment:
Thanks a lot for the changes, that we discussed beforehand. One small thing I would like to get changed, i.e., Since, you already put Hallucination Evaluation model inside `models/` folder, so can you import it here, instead of defining it here?

Reply:
Actually you are right, this shouldn't be here. We can now call it via scoring so why do I need this right? We have some duplication of score and metric. We need to take a deeper look at how to organize this. If we are doing score then do scores and not metrics. Or do we want to do both? 

@penguine-ip  your thoughts on this?

Reply:
Right now we want to keep both. The notion now is the following:

1. decouple the scoring logic
2. decouple the models as you mentioned. 
3. Metrics should call the function from the metrics/scoring.py to do all the computation.

Reply:
Yes so metrics will have `measure`, which imports a scorer from `scoring`. The main purpose of `measure` is to define what constitutes as a passing criteria for a given score (is it >=/=/<= minimum score, maximum score, etc), while the purpose of the scorer is to actually calculate the score itself. And of course if that requires a model we import it from `models`